### PR TITLE
feat(fault injection): change for diagnostic graph aggregator

### PR DESCRIPTION
**Pull request by KeisukeShima**
## Description

Because of the addition of a new diag collection node, [diagnostic_graph_aggregator](https://github.com/autowarefoundation/autoware.universe/tree/main/system/diagnostic_graph_aggregator), it is necessary to change the fault_injection mechanism.

The changes are as follows:
- Current spec (system_error_monitor): Diagnostics that contain part of the name are considered the same.
- New spec (diagnostic_graph_aggregator): only diagnostics with an exact name match are considered the same.

The sending rate of `/diagnostics` has been increased to 20Hz to comply with the new specification. Also, changed `diagnostic_updater` so that node names are not included in diagnostics names.

To maintain the current specifications after the change, the values in the configuration file have been modified.
I will create PR to autoware_launch to update configuration file as well.

## Tests performed

`system_error_monitor` works.

`diagnostic_graph_aggregator` works. (Blacked out areas are confidential.)
image (25)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TomohitoAndo** [comment]
@KeisukeShima 
Could you fix the pre-commit CI error?
https://results.pre-commit.ci/run/github/429651616/1712289373.eeutmg6_Q8G-UagZCrKmEw

**Comment by KeisukeShima** [comment]
@TomohitoAndo  All CI errors have been fixed.

**Comment by TomohitoAndo** [comment]
I set the config like following.
```
/**:
  ros__parameters:
    event_diag_list:
      /control/004-lane_departure: "lane_departure_checker_node: lane_departure"
      /control/005-trajectory_deviation: "lane_departure_checker_node: trajectory_deviation"
```

Execute this command.
```
$ ros2 topic pub /simulation/events tier4_simulation_msgs/msg/SimulationEvents "{fault_injection_events: [{name: /control/004-lane_departure, level: 2}]}"
```

I confirmed the diagnostics changed to ERROR  as expected.
image

**Review by TomohitoAndo — APPROVED** [review]
LGTM. Thanks!

File: simulator/fault_injection/src/fault_injection_node/fault_injection_node.cpp [hunk 1]
```diff
@@ -86,17 +86,17 @@ void FaultInjectionNode::updateEventDiag(
...
 {
   rcl_interfaces::msg::SetParametersResult result;
 
-  RCLCPP_DEBUG(this->get_logger(), "call onSetParam");
+  RCLCPP_DEBUG(this->get_logger(), "call on_set_param");
 
   try {
-    double value;
+    double value = NAN;
     if (tier4_autoware_utils::updateParam(params, "diagnostic_updater.period", value)) {
```
[the remark this answers is not in the document]
Comment (KeisukeShima) [reply simulator/fault_injection/src/fault_injection_node/fault_injection_node.cpp]: I think this code is not necessary. So I removed this function.
[6f46242](https://github.com/autowarefoundation/autoware.universe/pull/6750/commits/6f46242cca93003fd8395f5fcb37aafeea676ab8)

**Pull request merged**